src/sentry/search/events/builder/discover.py

```diff
@@ -583,13 +583,16 @@ def resolve_params(self) -> list[WhereType]:
         if self.end:
             conditions.append(Condition(self.column("timestamp"), Op.LT, self.end))
 
-        conditions.append(
-            Condition(
-                self.column("project_id"),
-                Op.IN,
-                self.params.project_ids,
+        # The if clause will prevent calling Snuba with an empty list of projects and complain with:
+        # sentry.utils.snuba.SnubaError: validation failed for entity events: missing required conditions for project_id
+        if self.params.project_ids:
```

Comment:
I put this change in #69499
I removed this change locally. I will push it tomorrow since I want to see all test results first.